### PR TITLE
Remove old Jalaali helpers in psammead-locales

### LINKED ISSUE #2355: Remove redundant Jalaali calendar logic
**Is your feature request related to a problem? Please describe.**
Once #2303 and [4122](https://github.com/bbc/simorgh/pull/4122) in Simorgh has been merged, we need to remove the old implementation for Jalaali calendar rendering for Pashto and Persian services.

**Describe the solution you'd like**
In `psammead-locales` the [jalaali helper](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-locales/moment/helpers/jalaali.js) along with its test file needs to be removed.

The Jalaali helper and its references need to be removed from the [`fa`](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-locales/moment/fa.js) and [`ps`](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-locales/moment/ps.js) locales.

**Note** we need to synchronise this removal with the addition of the new implementation in `psammead-timestamp-container` in order to avoid the user seeing duplicated timestamps.

**Testing notes**
Need to make sure that the tests for the [`fa`](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-locales/moment/fa.test.js) and [`ps`](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-locales/moment/ps.test.js) locales reflect the Jalaali helper removal.

Dev insight: Will there be any potential regression? etc

- [x] This feature is expected to need manual testing - e.g. adjusting UNIX timestamps and re-testing 'Updated' logic

**Additional context**

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Resolves #2355 

**Overall change:** Removed Jalaali and string helpers in psammead-locales.

**Code changes:**

- Removed Jalaali helper from psammead-locales
- Removed `fa` locale override as this was only adding the Jalaali calendar

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by oluoluoxenfree — APPROVED** [review]

**Review by RayNjeri — COMMENTED** [review]
Looks good so far but you may want to check on the drops on test coverage

**Comment by HarveyPeachey** [comment]
> Looks good so far but you may want to check on the drops on test coverage

Yeah the reason for those drops is that I've removed those files so code climate shouldn't really be erroring.

**Review by FK78 — APPROVED** [review]

File: packages/utilities/psammead-locales/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 | ------- | ----------- |
+| 2.24.0 | [PR#2375](https://github.com/bbc/psammead/pull/2300) Removed `fa` locale override and jalaali calendar helper  |
```
Comment (ryanmccombe) [inline packages/utilities/psammead-locales/CHANGELOG.md]:
```suggestion
| 2.24.0 | [PR#2375](https://github.com/bbc/psammead/pull/2375) Removed `fa` locale override and jalaali calendar helper  |
```

**Review by ryanmccombe — APPROVED** [review]

**Review by j-pendlebury — APPROVED** [review]

**Review by dr3 — CHANGES_REQUESTED** [review]
Chatted IRL, should be a breaking change :)

**Review by dr3 — APPROVED** [review]

**Comment by jamesdonoh** [comment]
Overriding CC total coverage change because it stems from the fact we are deleting code, not decreasing its coverage.

**Pull request merged**